scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -44,10 +48,17 @@ def process_response(self, request, response, spider):
                     # force recalculating the encoding until we make sure the
                     # responsetypes guessing is reliable
                     kwargs['encoding'] = None
+                if self.keep_encoding_header:
+                    if isinstance(kwargs.get('flags'), list):
```

Comment:
When I [said impossible](https://github.com/scrapy/scrapy/pull/4025#discussion_r328028801), I meant impossible. I think you may be mixing up `kwargs` and `response` here. Please, make sure you understand what is going on in this function.

Reply:
Its meant to process a response, ie decode and encoded input. If the request used to get the response was just for the 'HEAD' it finishes and the response that is the head is just returned. (Not 100% on what a 'HEAD' request is I'm assuming it like a header and contains general info on the actual site that a 'GET' could be sent to.

Then if it is a Response decode the response body into human readable(?) format. then create a another response, `respcls`, from the passed in response. Then create the kwargs dict with the fileds cls, class right?, as respcls and body taht is the decoded human readable format. Then if the created response, `respcls`, is a response that has a body of test add an additional entry for encoding that is None.

Then I added the if keep encoding header if where it checks if there is a kwargs entry for flags, an attribute that a typical Response object has correct or at least accumulates as it goes through the various filters and processing? Depending on, from the looks there wont be, if there are `flags` values add the flag 'decoded' 

Then to finalize it the original response has the values in kwargs replaced.......I'm not pulling the flags from the original response and then simple over riding them by creating the entry of flags with a single value erasing the accumulated flags from before aren't i?

After that clean up the headers but removing 'Content-Encoding' if that is desired. And ultimately return the processed and readable response object.